subprojects/core/src/main/java/org/gradle/process/internal/DefaultJavaExecAction.java

```diff
@@ -47,17 +48,35 @@
 public class DefaultJavaExecAction implements JavaExecAction {
 
     private final JavaExecHandleBuilder javaExecHandleBuilder;
-    private boolean ignoreExitValue;
+    private final Property<Boolean> ignoreExitValue;
+    private final Property<InputStream> standardInput;
+    private final Property<OutputStream> standardOutput;
+    private final Property<OutputStream> errorOutput;
 
-    public DefaultJavaExecAction(JavaExecHandleBuilder javaExecHandleBuilder) {
+    @Inject
+    public DefaultJavaExecAction(ObjectFactory objectFactory, JavaExecHandleBuilder javaExecHandleBuilder) {
         this.javaExecHandleBuilder = javaExecHandleBuilder;
+        this.ignoreExitValue = objectFactory.property(Boolean.class).convention(false);
+        this.standardInput = objectFactory.property(InputStream.class);
+        this.standardOutput = objectFactory.property(OutputStream.class);
+        this.errorOutput = objectFactory.property(OutputStream.class);
     }
 
     @Override
     public ExecResult execute() {
+        if (getStandardInput().isPresent()) {
+            javaExecHandleBuilder.setStandardInput(getStandardInput().get());
+        }
```

Comment:
❓ Why not use `getOrNull()` here and avoid the `if`?

Reply:
ExecHandle has defaults set that we don't want override. Also it throws exception if null standardInput stream is set

Reply:
Ah. :/ We should add a `Provider.ifPresent()` then... 🤔